File: Grammar1/Main.java
import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.tree.*;
import java.io.FileInputStream;
import java.io.IOException;

public class Main {
  public static void main(String args[]) throws IOException{
    // Reads from file
    FileInputStream input_stream = new FileInputStream(args[0]);
    CharStream input = CharStreams.fromStream(input_stream); 
    
    // Creates the Lexer
    G1Lexer lexer = new G1Lexer(input);
    CommonTokenStream tokens = new CommonTokenStream(lexer); 

    // Print the tokens readed
    tokens.fill();
    for(Token token : tokens.getTokens()){
        System.out.println(token.toString());
    }

    // Create the parser
    G1Parser parser = new G1Parser(tokens);
    ParseTree tree = parser.parse_all();

    if(parser.getNumberOfSyntaxErrors() == 0){
      System.out.println("Parsing result = SUCCESS");
      // Print the parsing tree
      System.out.println(tree.toStringTree(parser));
    }
  }
}

File: Grammar1/C++/main.cpp
#include <iostream>
#include <fstream>
#include <string>
#include "antlr4-runtime.h"
#include "G1Lexer.h"
#include "G1Parser.h"

using namespace antlr4;
using namespace std;

int main(int argc, const char* argv[]) {

  // Reads from file
  std::ifstream stream;                     // stream for input file
  stream.open(argv[1]);                     // open input file
  ANTLRInputStream input(stream);

  // Creates the Lexer
  G1Lexer lexer(&input);
  CommonTokenStream tokens(&lexer);

  // Print the tokens readed
  tokens.fill();
  for (auto token : tokens.getTokens()) {
    std::cout << token->toString() << std::endl;
  }
  std::cout << "\n";

  // Creates the Parser (parse_all is the starter rule)
  G1Parser parser(&tokens);
  tree::ParseTree *tree = parser.parse_all();

  if(parser.getNumberOfSyntaxErrors() == 0){
    std::cout << "Parsing result = SUCCESS" << std::endl;
    // Print the parsing tree
    std::cout << tree->toStringTree(&parser) << std::endl;
  }

  return 0;
}